### PR TITLE
Use Destination in Prometheus source

### PR DESCRIPTION
**Pull request by syedriko**
Fixes https://github.com/knative/eventing-contrib/issues/692

## Proposed Changes
The Prometheus Event Source now uses Destination as the type of Sink

### CODE REVIEW

File: prometheus/config/300-prometheussource.yaml [hunk 1]
```diff
@@ -70,14 +70,34 @@ spec:
...
+                        minLength: 1
+                      kind:
+                        type: string
+                        minLength: 1
+                      name:
+                        type: string
+                        minLength: 1
+                  uri:
+                    type: string
+                    description: "The target URI. If ref is provided, this must be a relative URI reference"
```
Comment (matzew) [inline prometheus/config/300-prometheussource.yaml]: Let's add `type` like done here: https://github.com/knative/eventing-contrib/pull/690/files

File: prometheus/pkg/reconciler/controller.go [hunk 2]
```diff
@@ -58,7 +59,7 @@ func NewController(
 		prometheusClientSet:    client.Get(ctx),
 	}
 	impl := controller.NewImpl(r, r.Logger, ReconcilerName)
-
+	r.sinkResolver = resolver.NewURIResolver(ctx, impl.EnqueueKey)
 	r.sinkReconciler = duck.NewSinkReconciler(ctx, impl.EnqueueKey)
```
Comment (matzew) [inline prometheus/pkg/reconciler/controller.go]: We do not need this `sinkReconciler` anymore: `sinkResolver` should be the only thing you need

File: prometheus/pkg/reconciler/prometheussource.go [hunk 3]
```diff
@@ -78,6 +79,7 @@ type Reconciler struct {
 
 	prometheusClientSet versioned.Interface
 
+	sinkResolver   *resolver.URIResolver
 	sinkReconciler *duck.SinkReconciler
```
Comment (matzew) [inline prometheus/pkg/reconciler/prometheussource.go]: similar: `sinkResolver` is what you need

**Comment by chizhg** [comment]
/home/prow/go/src/knative.dev/eventing-contrib is out of date. Please run hack/update-codegen.sh

File: prometheus/config/300-prometheussource.yaml [hunk 4]
```diff
@@ -70,14 +70,53 @@ spec:
...
+                    minLength: 1
+                  kind:
+                    type: string
+                    minLength: 1
+                  name:
+                    type: string
+                    minLength: 1
+                  uri:
+                    type: string
+                    description: "the target URI. If ref is provided, this must be relative URI reference."
```
Comment (mattmoor-sockpuppet) [inline prometheus/config/300-prometheussource.yaml]: Remove trailing whitespace:
```suggestion
                    description: "the target URI. If ref is provided, this must be relative URI reference."
```

**Pull request merged**